spirv_msl.cpp

```diff
@@ -2491,18 +2484,43 @@ string CompilerMSL::image_type_glsl(const SPIRType &type)
 	}
 
 	// Append the pixel type
-	auto &img_pix_type = get<SPIRType>(img_type.type);
 	img_type_name += "<";
-	img_type_name += type_to_glsl(img_pix_type);
+	img_type_name += type_to_glsl(get<SPIRType>(img_type.type));
 
-	if (img_type.is_written)
+	// For unsampled images, append the sample/read/write access qualifier.
+	// For kernel images, the access qualifier my be supplied directly by SPIR-V.
+	// Otherwise it may be set based on whether the image is read from or written to within the shader.
+	if (type.basetype == SPIRType::Image)
```

Comment:
Why are we not considering kernel mode?

Vulkan and SPIR-V support compute shaders, which I presume might get compiled from something like OpenCL. I understand that it might not be a strict requirement for regenerating GLSL...but I need to make sure `spirv_cross` can generate MSL can be used for Vulkan compute requirements in MoltenVK.